### DISCUSSION TITLE
Some default changes in Blink since v14, how can I change sources and copilot priority?

### DISCUSSION THREAD

**Discussion by banjo** [LazyVim/LazyVim#5091] [Q&A] [answered]
Hi,

I just updated to V14 and trying to make some minor changes, this is my current config:

```
return {
  "saghen/blink.cmp",
  opts = {
    keymap = { preset = "super-tab", ["<CR>"] = { "accept", "fallback" } },
    completion = {
      menu = {
        border = "rounded",
      },
      documentation = {
        window = {
          border = "rounded",
        },
      },
      ghost_text = {
        enabled = false,
      },
    },
    sources = {
      providers = {
        copilot = {
          name = "copilot",
          module = "blink-cmp-copilot",
        },
      },
      default = { "copilot", "lsp", "path" },
      cmdline = {},
    },
  },
}
```

There are some things I cannot get to work:

1. I only want to use the providers from lsp and path, I tried to only use those (and copilot) in the settings, but I still get things like this, string values that I don't care about. 

![CleanShot 2024-12-15 at 13 01 32@2x](https://github.com/user-attachments/assets/443512e2-eb78-4d27-bc6c-7752158cc678)

2. How do you guys work with copilot with blink? Currently, it is always at the bottom, so a bit cumbersome to always navigate to the bottom to see if there is a copilot suggestion. Can I put it on the top somehow or any other suggestions?

My old setup broke with cmp after updating this so would be nice to be able to make it work as good as possible!

**Comment by dpetka2001** [comment, marked as answer]
I believe that the following
```lua
return {
  "saghen/blink.cmp",
  opts = function(_, opts)
    opts.sources.default = { "copilot", "lsp", "path" }
  end,
}
```
should do what you want.

**Reply by banjo** [reply]
Thank you!

**Reply by ogios** [reply]
hi, i don't see any difference except changing from object to function, the sources in default are the same.
does this actually solve the `copilot priority` problem which op mentioned? i'm having the same `AI priority` problem

**Reply by dpetka2001** [reply]
This just ensures, that only these sources will be available like OP wanted. I don't know if the order matters for how `blink` will sort them like it did with `nvim-cmp`. Better ask at `blink` repo if there's something specific about sources sorting.
